app/src/app/admin/articles/article/article.ts

```diff
@@ -61,6 +65,30 @@ namespace app.admin.articles.article {
                         }
 
                         return articleService.getArticle($stateParams.permalink);
+                    },
+                    articleMetaTags: (article:common.models.Article):void => {
+
+                        // 'thru' not defined in typings yet
+                        article._articleMeta = (<any>_).chain(this.articleMetaTemplate)
+                            .map((metaTagName) => {
+                                let existingTag = _.find(article._articleMeta, {metaName:metaTagName});
+                                if(_.isEmpty(existingTag)) {
+                                    return new common.models.ArticleMeta({
+                                        metaName:metaTagName,
+                                        metaContent:''
+                                    });
+                                }
+                                return existingTag;
+                            })
+                            .thru((templateMeta) => {
+                                let leftovers = _.filter(article._articleMeta, (metaTag) => {
+                                    return !_.contains(templateMeta, metaTag);
+                                });
+
+                                return templateMeta.concat(leftovers);
+                            })
+                            .value();
+
```

Comment:
This resolve should either
a) Return the meta tags so the meta controller can edit them directly (assuming the binding still works)
b) If the binding doesn't work, move the logic to the article resolve function (probably create a static method in the article service to limit the logic that is in the article resolve. eg `hydrateMetaFromTemplate(meta:common.models.ArticleMeta[], template:string[]):meta:common.models.ArticleMeta`